pkg/handlers/nutanix/mutation/machinedetails/inject_worker.go

```diff
@@ -0,0 +1,153 @@
+	variableName      string
+	variableFieldPath []string
+}
+
+func NewWorkerPatch() *nutanixMachineDetailsWorkerPatchHandler {
+	return newNutanixMachineDetailsWorkerPatchHandler(
+		workerconfig.MetaVariableName,
+		v1alpha1.NutanixVariableName,
+		VariableName,
+	)
+}
+
+func newNutanixMachineDetailsWorkerPatchHandler(
+	variableName string,
+	variableFieldPath ...string,
+) *nutanixMachineDetailsWorkerPatchHandler {
+	return &nutanixMachineDetailsWorkerPatchHandler{
+		variableName:      variableName,
+		variableFieldPath: variableFieldPath,
+	}
+}
+
+func (h *nutanixMachineDetailsWorkerPatchHandler) Mutate(
+	ctx context.Context,
+	obj *unstructured.Unstructured,
+	vars map[string]apiextensionsv1.JSON,
+	holderRef runtimehooksv1.HolderReference,
+	_ client.ObjectKey,
+) error {
+	log := ctrl.LoggerFrom(ctx).WithValues(
+		"holderRef", holderRef,
+	)
+
+	nutanixMachineDetailsVar, found, err := variables.Get[v1alpha1.NutanixMachineDetails](
+		vars,
+		h.variableName,
+		h.variableFieldPath...,
+	)
+	if err != nil {
+		return err
+	}
+	if !found {
+		log.V(5).Info("Nutanix machine details variable for workers not defined")
+		return nil
+	}
+
+	log = log.WithValues(
+		"variableName",
+		h.variableName,
+		"variableFieldPath",
+		h.variableFieldPath,
+		"variableValue",
+		nutanixMachineDetailsVar,
+	)
+
+	return patches.MutateIfApplicable(
+		obj,
+		vars,
+		&holderRef,
+		selectors.InfrastructureWorkerMachineTemplates(
+			"v1beta1",
+			"NutanixMachineTemplate",
+		),
+		log,
+		func(obj *capxv1.NutanixMachineTemplate) error {
+			log.WithValues(
+				"patchedObjectKind", obj.GetObjectKind().GroupVersionKind().String(),
+				"patchedObjectName", client.ObjectKeyFromObject(obj),
+			).Info("setting Nutanix machine details in worker NutanixMachineTemplate spec")
+
+			obj.Spec.Template.Spec.BootType = capxv1.NutanixBootType(
+				nutanixMachineDetailsVar.BootType,
+			)
+			obj.Spec.Template.Spec.Cluster = capxv1.NutanixResourceIdentifier{
+				Type: capxv1.NutanixIdentifierType(nutanixMachineDetailsVar.Cluster.Type),
+			}
+			if nutanixMachineDetailsVar.Cluster.Type == v1alpha1.NutanixIdentifierName {
+				obj.Spec.Template.Spec.Cluster.Name = nutanixMachineDetailsVar.Cluster.Name
+			} else {
+				obj.Spec.Template.Spec.Cluster.UUID = nutanixMachineDetailsVar.Cluster.UUID
+			}
+
+			obj.Spec.Template.Spec.Image = capxv1.NutanixResourceIdentifier{
+				Type: capxv1.NutanixIdentifierType(nutanixMachineDetailsVar.Image.Type),
+			}
+			if nutanixMachineDetailsVar.Image.Type == v1alpha1.NutanixIdentifierName {
+				obj.Spec.Template.Spec.Image.Name = nutanixMachineDetailsVar.Image.Name
+			} else {
+				obj.Spec.Template.Spec.Image.UUID = nutanixMachineDetailsVar.Image.UUID
+			}
+
+			obj.Spec.Template.Spec.VCPUSockets = nutanixMachineDetailsVar.VCPUSockets
+			obj.Spec.Template.Spec.VCPUsPerSocket = nutanixMachineDetailsVar.VCPUsPerSocket
+			obj.Spec.Template.Spec.MemorySize = resource.MustParse(
+				nutanixMachineDetailsVar.MemorySize,
```

Comment:
Maybe we should use `resource.Quantity` in the CAREN API? I can pull it into https://github.com/d2iq-labs/cluster-api-runtime-extensions-nutanix/pull/13

Reply:
i tried that earlier but we want user to enter it as a string and then we convert that to resource.